### PR TITLE
Fixed port forward behaviour with single source port

### PR DESCRIPTION
**Pull request by MMaiero**
Signed-off-by: Maiero <matteo.maiero@eurotech.com>

### CODE REVIEW

File: kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java [hunk 1]
```diff
@@ -128,10 +130,27 @@ public PortForwardRule permittedMAC(String permittedMAC) {
     }
 
     public PortForwardRule sourcePortRange(String sourcePortRange) {
-        this.sourcePortRange = sourcePortRange;
+        parsePortRange(sourcePortRange);
         return this;
     }
 
+    private void parsePortRange(String sourcePortRange) {
```
Comment (pierantoniomerlino) [inline kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java:137]: Do we need some check on this field? For example that startPort <= endPort?

File: kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java [hunk 2]
```diff
@@ -384,7 +391,10 @@ public void setPermittedMAC(String permittedMAC) {
      * @return the sourcePortRange
      */
     public String getSourcePortRange() {
-        return this.sourcePortRange;
+        if (this.sourcePortStart == this.sourcePortEnd && this.sourcePortStart == 0) {
+            return null;
+        }
+        return this.sourcePortStart + ":" + this.sourcePortEnd;
     }
```
Comment (pierantoniomerlino) [inline kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java:398]: So, if there is only one source port, the result is port:port (i.e. 10100:10100). Is it correct?

File: kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java (hunk 1 above)
Comment (MMaiero) [reply kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java:137]: We could. But we don't have a way to report the exception. We could act safely by fixing something. The question is, what?

File: kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java (hunk 2 above)
Comment (MMaiero) [reply kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java:398]: yes

File: kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java (hunk 1 above)
Comment (pierantoniomerlino) [reply kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java:137]: Let's see it from another pov: what happens if we set a wrong range in the current implementation? Is there an exception when the rules is applied?

Comment (pierantoniomerlino) [reply kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java:137]: I verified that in the form in the web console there is a check for this...

Comment (MMaiero) [reply kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java:137]: True. But the code does not protect itself. For example in the case someone updates the configuration outside of the Web UI.
I think to do so we will need some breaking changes. But probably this is outside the scope of this PR.

Comment (pierantoniomerlino) [reply kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/iptables/PortForwardRule.java:137]: @MMaiero I totally agree. So, leave it as is and let's open an issue if needed.

**Review by pierantoniomerlino — APPROVED** [review]

**Pull request merged**